balancer_conn_wrappers.go

```diff
@@ -205,7 +205,7 @@ func (acbw *acBalancerWrapper) UpdateAddresses(addrs []resolver.Address) {
 	acbw.mu.Lock()
 	defer acbw.mu.Unlock()
 	if len(addrs) <= 0 {
-		acbw.ac.tearDown(errConnDrain)
```

Comment:
Should `tearDown` itself handle this case?

Reply:
It feels weird to me for addrConn to grab parent ClientConn's mu, and change ClientConn's fields directly (we do that for some of the keepalive configs today though).

I updated comment of `tearDown`, to direct callers to use `cc.removeAddrConn()` instead.

Reply:
That seems fine, I think it was a bad suggestion anyway for other reasons.